src/coreclr/jit/compiler.h

```diff
@@ -9593,6 +9638,36 @@ class Compiler
 
 #ifdef TARGET_XARCH
 public:
+
+    //------------------------------------------------------------------------
+    // compIsEvexOpportunisticallySupported - Checks for whether AVX10v1 or avx512InstructionSet is supported
+    // opportunistically.
+    //
+    // Returns:
+    //    returns true if AVX10v1 or avx512InstructionSet is supported opportunistically and
+    //    sets isV512Supported to true if AVX512F is supported, false otherwise.
+    //
+    bool compIsEvexOpportunisticallySupported(bool&                  isV512Supported,
+                                              CORINFO_InstructionSet avx512InstructionSet = InstructionSet_AVX512F)
+    {
+        // TODO: assert avx512InstructionSet is one of the AVX512 instruction sets
+        assert(avx512InstructionSet == InstructionSet_AVX512F || avx512InstructionSet == InstructionSet_AVX512F_VL ||
```

Comment:
Same question/comment as the above on whether we should just limit this to cases like `AVX512+VBMI` which aren't part of the EVEX baseline.

The consideration being that `compOpportunisticallyDependsOn` really exists to allow tracking when a given method is using an instruction that would otherwise cause the code to be unusable from a R2R/NAOT perspective.

The encoding for the instruction between `Avx10v1` and `Avx512` is then identical and so we don't actually care whether the CPU has `Avx512F_VL` or `Avx10v1` if they are using `vfixupimmps ymm, ymm, ymm/mem256`, we just care they have EVEX support.

The consideration then only comes in for 512-bit ISAs (where `vfixupimmps zmm, zmm, zmm/mem512` would get used and requires `Avx512F`) or for ISAs that aren't part of the "evex baseline" (like `Avx512Vbmi` which is part of `Avx10v1`, but not guaranteed as part of `Avx512`).

So what we really end up having is something like...
* `InstructionSet_EVEX` implies `Avx10v1` -or- `Avx512F+CD+DQ+BW+VL` and can be used for 128-bit or 256-bit EVEX encoding instructions in these namespaces
* `InstructionSet_Avx10v1_V512` should never be queried directly, it is covered by querying `Avx512*`
* `InstructionSet_Avx512F` is only queried for 512-bit EVEX encoded instructions
* `InstructionSet_Avx10v1` is only queried for cases like 128-bit/256-bit instructions that wouldn't be in `F+CD+DQ+BW+VL` (such as `VBMI`) and should appear with a corresponding query around `AVX512*_VL` (i.e. `AVX512_VBMI_VL`)

Reply:
To clarify, what we'd end up having is some mapping that handles these cases.

So a user may query:
```cpp
compIsEvexOpportunisticallySupported(&isV512Supported, InstructionSet_Avx512F)
```

and internally this would end up doing a `compOpportunisticallyDependsOn(InstructionSet_EVEX)` check and reporting whether that includes 512-bit support.

While if a user queried
```cpp
compIsEvexOpportunisticallySupported(&isV512Supported, InstructionSet_Avx512VBMI)
```

it would do the actual check against `avx512InstructionSet` vs `Avx10v1` like you have below.

Reply:
I have made some changes based on these comments. Let me know if they are enough to incorporate what you intended to explain in your comments.